pil-analyzer/src/evaluator.rs

```diff
@@ -535,7 +541,11 @@ mod internal {
                             if p == v {
                                 Some(value)
                             } else {
-                                match (p.try_to_integer(), v.clone().try_to_integer()) {
+                                // TODO I don't think this part is needed now that we have the type checker.
+                                match (
+                                    p.as_ref().clone().try_to_integer(),
+                                    v.as_ref().clone().try_to_integer(),
```

Comment:
Maybe `try_to_integer()` should take `&self` instead of `self`, so no cloning would be needed here.

Reply:
Thought about that already as well...
Will check how ofter `try_to_integer` would needlessly clone in that case, but I guess almost never.